lib/apollo-exception.filter.ts

```diff
@@ -0,0 +1,39 @@
+import { Catch, ArgumentsHost, HttpException } from '@nestjs/common';
+import {
+  ApolloError,
+  AuthenticationError,
+  ForbiddenError,
+} from 'apollo-server-errors';
+import { GqlExceptionFilter } from './interfaces';
+import { GqlContextType } from './services';
+
+const apolloPredefinedExceptions: Record<number, typeof ApolloError> = {
+  401: AuthenticationError,
+  403: ForbiddenError,
+};
+
+@Catch(HttpException)
+export class ApolloExceptionFilter implements GqlExceptionFilter {
+  catch(exception: HttpException, host: ArgumentsHost) {
+    if (host.getType<GqlContextType>() !== 'graphql') {
+      throw exception;
```

Comment:
Is it ok to throw an exception here? From what I understand it not ok

> The ExceptionFilter is always the last place that gets called before a response is sent out, it is responsible for building the response. You cannot rethrow an exception from within an ExceptionFilter.

Since this PR was not merged I copied this class to our own project. Our project has both a graphQL and REST routes. This line is crashing the App. I had to make the following change to make it work 

```javascript
 if (host.getType<GqlContextType>() !== 'graphql') {
      const ctx = host.switchToHttp();
      const response = ctx.getResponse<Response>();
      const status = exception.getStatus();

      response
        .status(status)
        .json({
          statusCode: status,
          message: exception.message
        });
      return exception;
    }
```

I am still new to NestJS. SO am I missing something here?

Reply:
You're right, thanks for flagging! I've changed it though I preferred to inherit from `BaseExceptionFilter` and just call its `.catch` to keep the logic consistent with the rest of Nest.